### PR TITLE
Feat(deactivate-all-county-users)

### PR DESCRIPTION
**Pull request by Noam6421**
This PR implements feature [1485](https://dev.azure.com/spectrumFactory/CoronaI/_sprints/backlog/CoronaI%20Team/CoronaI/Sprint%2014?workitem=1485)

In this PR I added deactivation button that appear only to users from type ADMIN and that when clicked updates all the users of the current county activity statuses to false.
Added new route for that purpose and new DB function.

### CODE REVIEW

File: client/src/components/App/Content/UsersManagement/useUsersManagement.ts [hunk 1]
```diff
@@ -264,6 +269,33 @@ const useUsersManagement = ({ page, rowsPerPage, cellNameSort, setPage }: useUse
...
+        }).then((result) => {
+            if (result.value) {
+                const deactivateAllCountyUsersLogger = logger.setup('Updating all county users activity status to false');
+                deactivateAllCountyUsersLogger.info('send request to server for updating users activity statuses', Severity.LOW);
+                setIsLoading(true);
+                return axios.post('users/deactivateAllCountyUsers', {
+                    countyId: displayedCounty
+                }).then((result) => {
+                    if(result.data)
+                        fetchUsers();
```
Comment (y-golde) [inline client/src/components/App/Content/UsersManagement/useUsersManagement.ts:288]: add { }

File: server/src/ClientToDBAPI/UsersRoute/mainRoute.ts [hunk 2]
```diff
@@ -140,6 +140,25 @@ usersRoute.post('/updateIsUserActiveById', adminMiddleWare, (request: Request, r
     updateIsUserActive(response, request.body.userId, request.body.isActive);
 })
 
+usersRoute.post('/deactivateAllCountyUsers', (request: Request, response: Response) => {
+    const deactivateAllCountyUsers = logger.setup({
+        workflow: 'updating county users activity status to false',
+        user: response.locals.user.id,
+    });    
+    const parameters = { countyId: request.body.countyId }
```
Comment (y-golde) [inline server/src/ClientToDBAPI/UsersRoute/mainRoute.ts:148]: change to request.body.county (for the future middleware)

**Review by y-golde — CHANGES_REQUESTED** [review]

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx [hunk 3]
```diff
@@ -216,18 +220,32 @@ const UsersManagement: React.FC = () => {
...
-                            variant='dot'
-                            anchorOrigin={{ horizontal: 'left', vertical: 'top' }}
+
+                <Grid item>
+                    {
+                        (userType === UserType.ADMIN) &&
+                        <Button
+                            variant='contained'
+                            color='inherit'
+                            className={classes.deactivateButton}
```
[the remark this answers is not in the document]
Comment (Noam6421) [reply client/src/components/App/Content/UsersManagement/UsersManagement.tsx:230]: its purple

File: client/src/components/App/Content/UsersManagement/UsersManagementStyles.ts [hunk 4]
```diff
@@ -51,7 +51,15 @@ const useStyles = makeStyles((theme: Theme) => ({
     desks: {
         width: '9vw',
         height: '2vw'
+    },
+    deactivateButton: {
+        borderRadius: 48,
```
[the remark this answers is not in the document]
Comment (Noam6421) [reply client/src/components/App/Content/UsersManagement/UsersManagementStyles.ts:56]: px , added  it

File: server/src/ClientToDBAPI/UsersRoute/mainRoute.ts (hunk 2 above)
Comment (Noam6421) [reply server/src/ClientToDBAPI/UsersRoute/mainRoute.ts:148]: changed that

File: client/src/components/App/Content/UsersManagement/UsersManagement.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (y-golde) [reply client/src/components/App/Content/UsersManagement/UsersManagement.tsx:230]: whoops

**Review by y-golde — APPROVED** [review]

**Review by chenFriedman — APPROVED** [review]

**Pull request merged**